### ISSUE TITLE
No support for inhertiance or generic parameters

### ISSUE THREAD

**Issue by Sshnyari** [thomas-girotto/MockGen#1]
Using source generators like this is great and the way you did the mocking is intuitive and not complicated to use.
Unfortunately, I couldn't use it in the first project I tried it on. I encountered two issues.
Let's say we have these two interfaces we want to mock

```
    public interface IMockableTest
    {
        T GenericMethod1<T>();
        void GenericMethod2<T>(T value);
        bool Method3(string value);
    }

    public interface IMockableTest2 : IMockableTest
    {

    }

```

If I try to mock IMockableTest with `var mock = MockG.Generate<IMockableTest>().New();,` I get these errors
![image](https://user-images.githubusercontent.com/16015858/123139242-4bdd4380-d456-11eb-99fd-83ab96b78164.png)

If I try to mock IMockabkeTest2 with `var mock = MockG.Generate<IMockableTest2>().New();,`, I get these errors
![image](https://user-images.githubusercontent.com/16015858/123139469-8810a400-d456-11eb-8bee-89685a2a68bf.png)
(and obviousely I cannot mock the parent class methods)

**Comment by thomas-girotto** [comment]
Yep, you're right, i didn't think about testing generic methods while developping my lib, and i realized it doesn't work some weeks ago while trying to use it on another project.

I'm working to add support for them, though i still didn't find the right way to do it yet. Unless i'm missing some clever way to do it, it requires quite a lot of changes...

As for inheritance, thanks for the feedback, i wasn't aware it doesn't work either. This use case should be a lot easier to handle though.

Thanks for the first issue :)
